packages/protocol-perf/src/index.ts

```diff
@@ -52,12 +52,19 @@ import type { AbortOptions } from '@libp2p/interfaces'
 const log = logger('libp2p:perf')
 
 export const defaultInit: PerfServiceInit = {
-  protocolName: '/perf/1.0.0',
+  protocolName: '/perf/1.2.0',
```

Comment:
There is no change on the network layer, right? Why bump the minor version?

Reply:
I am wondering if [the spec](https://github.com/libp2p/specs/blob/master/perf/perf.md#protocol) should be amended to address the scenario where we are not able to send over the `MAX_INTEGER` amount of bytes in the 60s timeframe.

Step 2 in the client workflow states:

```
2. Tell the server how many bytes we want the server to send us as a single big-endian uint64 number. Zero is a valid number, so is the max uint64 value.
```

In this case we are requesting an _unlimited server response_

Step 5 states:

```
5. Read from the read side of the stream. This should be the same number of bytes as we told the server in step 2.
```

If we had requested an _infinite_ amount of bytes, would read side match that amount requested?

Reply:
> Step 2 in the client workflow states:
> 
> ```
> 2. Tell the server how many bytes we want the server to send us as a single big-endian uint64 number. Zero is a valid number, so is the max uint64 value.
> ```
> 
> In this case we are requesting an _unlimited server response_

Correct. In other words the server keeps sending data for max uint64 and given that that is such a large number, it never finishes.

> Step 5 states:
> 
> ```
> 5. Read from the read side of the stream. This should be the same number of bytes as we told the server in step 2.
> ```
>
> If we had requested an infinite amount of bytes, would read side match that amount requested?

Given that max uint64 is so large, the transfer will not finish within 20s, the client will be killed (through Linux `timeout` tool) and thus the connection reset on the server side. Neither the client nor the server is aware of the timeout (20s), but the client is simply killed by the benchmark runner.

Does that help?

Reply:
> Given that max uint64 is so large, the transfer will not finish within 20s, the client will be killed (through Linux timeout tool) and thus the connection reset on the server side. Neither the client nor the server is aware of the timeout (20s), but the client is simply killed by the benchmark runner.

I understand this execution flow., but my question is does Step 5 in the spec contradict this? Because the client is killed before the _infinite_ amount of bytes would be read.

Reply:
In the throughput benchmark the client is killed before step 5, i.e. it is killed before it can finish reading all data from the stream and before it can check whether it received enough data. Thus the client will never print the `final` output, but only the `intermediary` output. That is fine as the throughput benchmark only needs the latter, not the former.

For the latency benchmark, the client is not killed, but able to finish the whole flow, including step 5, given that only a small amount (1 byte) is uploaded and downloaded. Thus in this benchmark the client will print the `final` output.

Does that make sense @maschad?

Reply:
That makes sense. Thanks for clarifying!